src/System.Management.Automation/engine/remoting/fanin/PriorityCollection.cs

```diff
@@ -154,15 +162,29 @@ internal void Add<T>(RemoteDataObject<T> data)
         /// </summary>
         internal void Clear()
         {
-            Dbg.Assert(null != _dataToBeSent, "Serialized streams are not initialized");
-            lock (_syncObjects[(int)DataPriorityType.PromptResponse])
+            /* Lock _dataSyncObjects and _dataToBeSent as a set to ensure atomic cleanup. */
+            lock (_syncObject)
             {
-                _dataToBeSent[(int)DataPriorityType.PromptResponse].Dispose();
-            }
+                /*
+                    NOTE: Error paths during initialization can cause _dataSyncObjects to be null
+                    causing an unhandled exception in finalize and a process crash.
+                */
+                if (null != _dataSyncObjects)
+                {
+                    Dbg.Assert(null != _dataToBeSent, "Serialized streams are not initialized");
```

Comment:
This assert should be removed because apparently `_dataToBeSent` can be null in that error path.

Reply:
Never mind this comment. It's certainly right to assume `_dataToBeSent != null` when `null != _dataSyncObjects` 😄

Reply:
Will do.

Reply:
fixed